FRW/Modules/Login/ViewModel/SeedPhraseLoginViewModel.refactored.swift

```diff
@@ -0,0 +1,161 @@
+
+        // Validate each word
+        for word in words {
+            if Mnemonic.search(prefix: String(word)).isEmpty {
+                hasError = true
+                break
+            }
+        }
+
+        let valid = Mnemonic.isValid(mnemonic: original)
+
+        // Update suggestions
+        if text.last == " " || valid {
+            suggestions = []
+        } else {
+            suggestions = Mnemonic.search(prefix: String(words.last ?? ""))
+        }
+
+        updateState()
+    }
+
+    func updateState() {
+        if isAdvanced {
+            buttonState = words.isEmpty || derivationPath.isEmpty ? .disabled : .enabled
+        } else {
+            buttonState = words.isEmpty ? .disabled : .enabled
+        }
+    }
+
+    func onAdvance() {
+        isAdvanced.toggle()
+        updateState()
+    }
+
+    // MARK: - LoginViewModelProtocol Required Methods
+
+    func onSubmit() {
+        UIApplication.shared.endEditing()
+
+        let rawMnemonic = words.condenseWhitespace()
+
+        Task {
+            // Validate HD wallet
+            guard let hdWallet = HDWallet(mnemonic: rawMnemonic, passphrase: passphrase) else {
+                HUD.error(title: "invalid_data".localized)
+                return
+            }
+
+            // Validate advanced options
+            if isAdvanced && derivationPath.isEmpty {
+                HUD.error(title: "required_info_not".localized)
+                return
+            }
+
+            // Create seed phrase key
+            if isAdvanced && !derivationPath.isEmpty {
+                cryptoKey = FlowWalletKit.SeedPhraseKey(
+                    hdWallet: hdWallet,
+                    storage: FlowWalletKit.SeedPhraseKey.seedPhraseStorage,
+                    derivationPath: derivationPath,
+                    passphrase: passphrase
+                )
+            } else {
+                cryptoKey = FlowWalletKit.SeedPhraseKey(
+                    hdWallet: hdWallet,
+                    storage: FlowWalletKit.SeedPhraseKey.seedPhraseStorage
+                )
+            }
+
+            guard cryptoKey != nil else {
+                HUD.error(title: "invalid_data".localized)
+                return
+            }
+
+            // Create wallet with seed phrase key
+            wallet = FlowWalletKit.Wallet(
+                type: .key(cryptoKey!),
+                networks: [currentNetwork]
+            )
+
+            HUD.loading()
+
+            // Fetch all addresses (common logic from protocol)
+            try await fetchAllAddresses()
+
+            HUD.dismissLoading()
+
+            // Select account (common logic from protocol)
+            await selectAccountFromWallet()
+        }
+    }
+
+    func getP256PublicKey() -> String? {
+        cryptoKey?.publicKey(signAlgo: .ECDSA_P256)?.hexValue.dropPrefix("04")
+    }
+
+    func getSecp256PublicKey() -> String? {
+        cryptoKey?.publicKey(signAlgo: .ECDSA_SECP256k1)?.hexValue.dropPrefix("04")
```

Comment:
check the length first, don't blind remove 04 prefix, we had this issue before. If user's public key start with `0404`, we might remove the wrong prefix